api/controllers/MerchStoreController.ts

```diff
@@ -235,4 +239,18 @@ export class MerchStoreController {
       editOrderPickupEventRequest.pickupEvent);
     return { error: null, pickupEvent: pickupEvent.getPublicOrderPickupEvent() };
   }
+
+  @Get('/store/cart')
+  async getCart(@QueryParams() getCartRequest:GetCartRequest,
+    @AuthenticatedUser() user: UserModel): Promise<GetCartResponse> {
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    const items = await this.merchStoreService.getCartItems(getCartRequest.items);
+    const foundItems = Array.from(items.values())
+      .map((o) => o.uuid);
+    if (foundItems !== getCartRequest.items) {
+      const missingItems = difference(getCartRequest.items, foundItems);
+      throw new NotFoundError(`The following items were not found: ${missingItems}`);
+    }
```

Comment:
Instead of returning the `missingItems` list within the string of the error message, we should instead return it in some kind of JSON so that the frontend can easily get the list of missing items without doing any string manipulation. But it seems like we can't pass in the JSON data separately as a constructor to `NotFoundError`. @sumeet-bansal what you recommend here? Since frontend would need to use the list of missing items to clear the missing items from its own local storage, it'd need to parse the list somehow